test/map_node_test.cpp

```diff
@@ -147,3 +147,12 @@ TEST(MapNode, sixty_six_to_ninety_eight_water_border) {
 		EXPECT_EQ(next_water_border_itr, node_98.neighbors().end());
 	}
 }
+
+TEST(MapNode, is_not_adjacent_to_self) {
+	wl::MapNode::ID i{ 0u };
```

Comment:
can we name `i` a little more descriptively? I'm thinking like `self_id` or something